app/mixins/check-authenticated-route-mixin.js

```diff
@@ -0,0 +1,8 @@
+import Mixin from '@ember/object/mixin';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
+export default Mixin.create(AuthenticatedRouteMixin, {
```

Comment:
Why do you need a mixin for that?

Reply:
To avoid duplicating of following in your future routes:
```
authenticationRoute: 'login',
routeAfterAuthentication: 'protected',
routeIfAlreadyAuthenticated: 'protected'
```
It's easier to have these settings in mixin